### PR TITLE
Add code from omnibus 3 to retry failed downloads

### PR DESCRIPTION
**Pull request by Hobbsee**
The omnibus-3 branch has retry code for failing downloads.  This is very helpful for imperfect networks, and it would be great to have this in version 4 as well.

### CODE REVIEW

**Comment by Hobbsee** [comment]
@schisamo , did you have an update on this?

**Comment by schisamo** [comment]
@Hobbsee Overall this looks great! Could we get some unit test coverage in https://github.com/opscode/omnibus/blob/master/spec/unit/fetchers/net_fetcher_spec.rb

**Comment by Hobbsee** [comment]
@schisamo Certainly.  Tests are added via Webmock to simulate the failure, and a logic error has been fixed.

**Comment by schisamo** [comment]
@Hobbsee Looking at the Travis CI failure it looks like Webmock is causing some issues with other specs.

**Comment by Hobbsee** [comment]
@schisamo Well, that's embarrassing!  They were working for me locally with bundle exec rspec, but the rake task was failing.  Sorry about that!

I've now fixed that up, verified that the tests are working via rake, and they're also running via travis.

File: lib/omnibus/generator_files/omnibus.rb.erb [hunk 1]
```diff
@@ -40,6 +40,7 @@
 # solaris_compiler 'gcc'
 # build_retries 5
 # fetcher_read_timeout 120
+# fetcher_retries 120
```
Comment (schisamo) [inline lib/omnibus/generator_files/omnibus.rb.erb]: Probably just a copy/pasta error but we want a slightly smaller default here. :)

**Comment by Hobbsee** [comment]
@schisamo an excellent point.  This has been updated.

**Comment by schisamo** [comment]
@Hobbsee So I think one more rebase onto current master and this branch will be ready to merge!

**Comment by Hobbsee** [comment]
@schisamo Excellent!  This has been rebased.

**Comment by schisamo** [comment]
Thanks @Hobbsee! This was merged as: https://github.com/chef/omnibus/compare/ed0e77e06ecc6adf24436f6d2e940ad9b5032262...78db0d50412ec6e0cc2fbae29064674cf082fce3

I did a proper rebase onto master (vs merges of master back into the feature branch). I also squashed a few commits down. And of course, all author attribution was still yours!

I also made the `WebMock.allow_net_connect!` global: https://github.com/chef/omnibus/commit/4c4ff73b8c7c89c2b7a5b52dea7d847a8365118c

**Comment by Hobbsee** [comment]
@schisamo Lovely, thanks!

**Pull request closed without merging**